app/dashboard/views.py

```diff
@@ -63,6 +63,36 @@ def send_tip(request):
     return TemplateResponse(request, 'yge/send1.html', params)
 
 
+def record_user_tip_action(profile_handle, event_name, tip_pk):
```

Comment:
Could we get away with something like this?

```python
def record_user_action(profile_handle, event_name, instance):
    instance_class = instance.__class__.__name__.lower()

    try:
        user_profile = Profile.objects.get(handle=profile_handle)
        UserAction.objects.create(
            profile=user_profile,
            action=event_name,
            metadata={
                f'{instance_class}_pk': instance.pk,
            })
    except Exception as e:
        # TODO: sync_profile?
        logging.error(f"error in record_action: {e} - {event_name} - {instance}")
```

Reply:
hows this? 5a44b0d81dac184b93a731d37c5fc10022fbaa29